packages/fast-components-react-msft/src/badge/badge.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    BadgeAppearance,
+    BadgeHandledProps,
+    BadgeProps,
+    BadgeSize,
+    BadgeUnhandledProps,
+} from "./badge.props";
+import { get } from "lodash-es";
+
+class Badge extends Foundation<BadgeHandledProps, BadgeUnhandledProps, {}> {
+    public static displayName: string = "Badge";
+
+    public static defaultProps: Partial<BadgeProps> = {
+        appearance: BadgeAppearance.lowlight,
+        size: BadgeSize.small,
+    };
+
+    protected handledProps: HandledProps<BadgeHandledProps> = {
+        appearance: void 0,
+        size: void 0,
+        managedClasses: void 0,
+    };
+
+    /**
+     * Renders the component
+     */
+    public render(): JSX.Element {
+        return (
+            <span {...this.unhandledProps()} className={this.generateClassNames()}>
+                {this.props.children}
+            </span>
+        );
+    }
+
+    /**
+     * Generates class names
+     */
+    protected generateClassNames(): string {
+        let classNames: string = get(this.props, "managedClasses.badge", "");
+
+        switch (this.props.appearance) {
+            case BadgeAppearance.highlight:
+                classNames = `${classNames} ${get(
+                    this.props,
+                    "managedClasses.badge__highlight",
+                    ""
+                )}`;
+                break;
+            case BadgeAppearance.lowlight:
+                classNames = `${classNames} ${get(
+                    this.props,
+                    "managedClasses.badge__lowlight",
+                    ""
+                )}`;
+                break;
+            case BadgeAppearance.accent:
```

Comment:
This switch never checks for lightweight, why do we have it as part of BadgeAppearance?

Reply:
"Lightweight" is just a badge, with no additional class name added. In the editor, you would switch to lightweight to see that appearance, hence it being in appearance.

I think this will change though based on @chrisdholt 's comments

Reply:
I think we should either add the class here or make the enum optional. In my opinion the default is just the default and the enums are modifiers. This is really just tricking the system. I think if we make this optional then things work as expected. Default badge with no modifier has a lightweight appearance.

Reply:
Removing lightweight